release-notes/Cabal-3.12.0.0.md

```diff
@@ -0,0 +1,258 @@
+
+    Hie files generated by GHC are now stored in the
+    `extra-compilation-artifacts` directory which gets installed with the
+    package.
+
+- Include the GHC "Project Unit Id" in the cabal store path [#8114](https://github.com/haskell/cabal/issues/8114) [#9326](https://github.com/haskell/cabal/pull/9326)
+
+    This allows the use of several **API incompatible builds of the same
+    version of GHC** without corrupting the cabal store.
+
+    This relies on the "Project Unit Id" which is available since GHC 9.8.1,
+    older versions of GHC do not benefit from this change.
+
+- Fix the platform string for GNU/Hurd [#9434](https://github.com/haskell/cabal/pull/9434)
+
+    Depending whom you ask, GNU/Hurd will be labelled "gnu" or "hurd". The
+    autotools use "gnu", so ghc follows this for installed files, even if the
+    ghc source code uses OSHurd. We thus need to add the translation between
+    the two.
+
+- Use linker capability detection to improve linker use [#9443](https://github.com/haskell/cabal/pull/9443)
+
+    Previously the GHC version number and platform were used as a proxy for
+    whether the linker can generate relocatable objects.
+
+    Now, the ability of the linker to create relocatable objects is detected.
+
+- Merge globbing implementations [#5349](https://github.com/haskell/cabal/issues/5349) [#9673](https://github.com/haskell/cabal/pull/9673)
+
+    The common aspects of the globbing functionality between `Cabal` and
+    `cabal-install` have been factored out. The only change in the user-facing
+    API is that we now record when a glob does not match exactly, but matches
+    a directory with that same name, with the new constructor
+    `GlobMatchesDirectory` of `GlobResult`.
+
+    To illustrate, this change means that when `foo/dir` is a directory, the
+    glob `*/dir/` matches exactly `foo/dir` (as before), but now `*/dir`
+    produces `GlobMatchesDirectory` instead of failing.  This allows callers
+    to decide whether to allow or discard such inexact matches.
+
+- Document `remote-repo-cache` as implemented. [#8737](https://github.com/haskell/cabal/issues/8737) [#8738](https://github.com/haskell/cabal/pull/8738)
+- Deduplicate LD_LIBRARY_PATH when running tests [#8728](https://github.com/haskell/cabal/pull/8728)
+- Add support for the 64-bit RISC-V architecture [#9062](https://github.com/haskell/cabal/pull/9062)
+- Add support for 64-bit LoongArch architecture [#9215](https://github.com/haskell/cabal/pull/9215)
+- Don't report `index.html` file as created, if not created by Haddock [#5120](https://github.com/haskell/cabal/issues/5120) [#9332](https://github.com/haskell/cabal/pull/9332)
+- Enable using $ORIGIN in RPATH on GNU/Hurd [#9441](https://github.com/haskell/cabal/pull/9441)
+- Add support for 64-bit SPARC as a separate architecture [#9445](https://github.com/haskell/cabal/pull/9445)
```

Comment:
Maybe grouping all new architectures under a single bullet list and having a sublist with dense presentation (one item pre architecture, no empty lines between the items) would look better than switching back and forth between dense and sparse presentations of the items in the top-level list.